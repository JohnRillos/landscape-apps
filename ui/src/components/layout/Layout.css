:root {
  --header-height: 80px;
  --aside-width: 120px;
}

.layout {
  display: grid;
  grid-template-areas:
    'header header'
    'aside main'
    'footer footer';
  grid-template-columns: auto 1fr; /** collapsible aside, main expands to fill */
  grid-template-rows: auto minmax(0, 1fr) auto; /** collapsible header, main expands to fill, collapsible footer */
  height: 100%;
}

.aside {
  grid-area: aside;
  width: var(--aside-width);
}

.header {
  grid-area: header;
  height: var(--header-height);
}

.footer {
  grid-area: footer;
}

.main {
  grid-area: main;
}
