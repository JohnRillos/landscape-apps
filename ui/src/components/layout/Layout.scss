$header-height: 80px;
$footer-height: 80px;
$sidebar-width: 120px;

div#layout {
  display: grid;
  grid-template-areas:
  "header header"
  "sidebar main"
  "footer footer";
  grid-template-columns: minmax(min-content, $sidebar-width) auto; /** collapsible sidebar, main content */
  grid-template-rows: minmax(max-content, $header-height) auto minmax(max-content, $footer-height); /** collapsible header, auto main, collapsible footer */
  
  div#sidebar {
    grid-area: sidebar;
  }

  div#header {
    grid-area: header;
  }

  div#footer {
    grid-area: footer;

  }

  div#main {
    grid-area: main;
  }
}
