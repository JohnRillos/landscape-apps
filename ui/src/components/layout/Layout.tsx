import React from 'react';

import './Layout.scss';

interface LayoutProps {
  children: React.ReactNode;
  footer?: React.ReactNode;
  header?: React.ReactNode;
  sidebar?: React.ReactNode;
}

export default function Layout({ children, footer, header, sidebar }: LayoutProps) {
  return (
    <div id="layout">
      <div id="header">{header}</div>
      <div id="sidebar">{sidebar}</div>
      <div id="main">{children}</div>
      <div id="footer">{footer}</div>
    </div>
  );
}
