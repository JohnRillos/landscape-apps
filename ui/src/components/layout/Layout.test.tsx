/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { describe, expect, it } from 'vitest';
import renderer from 'react-test-renderer';
import Layout from './Layout';

describe('Layout', () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(<Layout main={<p>content</p>}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
