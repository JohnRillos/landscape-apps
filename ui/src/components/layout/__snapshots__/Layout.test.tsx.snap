// Vitest Snapshot v1

exports[`Layout > renders as expected 1`] = `
<div
  id="layout"
>
  <div
    id="header"
  />
  <div
    id="sidebar"
  />
  <div
    id="main"
  >
    <p>
      content
    </p>
  </div>
  <div
    id="footer"
  />
</div>
`;
